src/main/java/com/devonfw/ide/sonarqube/common/impl/check/naming/DevonNamingConventionClassExtendsClassUcImplCheck.java

```diff
@@ -55,14 +55,12 @@ protected boolean checkClassNameAndCreateIssue(JavaFileScannerContext context) {
    */
   protected boolean isImplementingCorrectInterface(String superInterface) {
 
+    boolean isImplementingCorrectInterface = false;
+
     for (String name : this.superInterfacesNames) {
-      if (superInterface.equals(name)) {
-        return true;
-      } else {
-        return false;
-      }
+      isImplementingCorrectInterface = superInterface.equals(name);
```

Comment:
actually this was wrong before and still is wrong.
If `superInterface.equals(name)` is `true` this method should return true. Currently it only does if the expected interface is the last one.